### DISCUSSION TITLE
hole

### DISCUSSION THREAD

**Discussion by eskan** [ph1p/ikea-led-obegraensad#122] [Ideas]
Hi everyone, I just finished mine and wanted to thank everyone who create/participate to that hack.
Removing the back was a bit tricky, so if I ever make another one, I think I'll cut a rectangular hole and then 3D print a cover for it.
Cheers,
Cyrille.

**Comment by JanPetterMG** [comment]
Best **non-destructive** solution is to use an ~3 mm~ **2,5 mm drill bit**, then it's quite easy, and only takes a minute or two. Smaller may work, but bigger will definitively harm the frame permanently.
Of course, you won't be able to re-use the pop-nails, but nothing else will be destroyed in the process. For those who want to, they can buy new nails, and practically speaking restore it to it's original state, or just fastening it with screws instead. Your choice.

Side note:
Had some trouble on some animations, random pixels near the bottom getting stuck in their previous state from time to time, but I solved it by installing a capacitor and adding some extra wiring. Not sure if both are necessary, but it effectively eliminated the problem.

I've got mine just 2 weeks ago, but I've already fixed lots of bugs, cleaned up the code, as well as implemented new features and a number of new plugins. Will probably post it as an fork in a week or two, as the owner seemingly has lost interest in the project. It's more like a version `1.9`, so I don't expect it ever to be merged, given the state of existing PR's have died out.

I've also got it working on newer type of boards. ESP32-C3 for example wouldn't compile at all, but is now handling it all strong.

**Reply by c64emulator** [reply]
> Best **non-destructive** solution is to use an **3 mm drill bit**, then it's quite easy, and only takes a minute or two. Smaller may work, but bigger will definitively harm the frame permanently. Of course, you won't be able to re-use the pop-nails, but nothing else will be destroyed in the process. For those who want to, they can buy new nails, and practically speaking restore it to it's original state, or just fastening it with screws instead. Your choice.

I used a 6mm drill. This diameter will simply cut of the head of the washouts. I did never drill through / into the back of the box.
In order to prevent falling small metal parts into the box (dangerous for electricity), I used some tape for all open mounting holes  and a vacuum cleaner while drilling.
I closed the housing with screws "Knauf Blechschraube LN 3,5 mm x 9 mm", EAN 4006379079032 
![KNAUF Schnellbauschrauben LN 3 5 mm x 9 mm-02](https://github.com/user-attachments/assets/a950b473-5ebe-44ba-a79a-f5e6029f55e5)
This screw is available in many DIY-stores all over germany.
![IKEA-Hack - Wandleuchte OBEGRÄNSAD -03](https://github.com/user-attachments/assets/9011f919-952b-479a-8917-a48407494638)

**Reply by c64emulator** [reply]
> Side note: Had some trouble on some animations, random pixels near the bottom getting stuck in their previous state from time to time, but I solved it by installing a capacitor and adding some extra wiring. Not sure if both are necessary, but it effectively eliminated the problem.

I designed a small PCB with fritzing.org, added a 100nF, a 100uF and a LDR. For future use I have also an additional GPIO on AX1  (for e.g. a buzzer or microphone), a picoblade connector for I2C (for e.g. RTC or sensors) and a picoblade connector for serial monitoring.
<img width="861" height="625" alt="v7_7" src="https://github.com/user-attachments/assets/a7ca9584-863e-49a6-a88c-aa64d6469a22" />
![bestueckt-v7](https://github.com/user-attachments/assets/dc347003-2e93-4e14-a1fc-f8d28acb6ae5)

The PCB is connected to the LED panel using JST-XH connectors. For J1 I had to swap the location of the red wire with the green wire on a standard 6-pin-connector from bay. As the two connections on the right side are both 2-wired I used one 3-pin-connector for J4 to prevent mixup. 
![IKEA-Hack - Wandleuchte OBEGRÄNSAD -06](https://github.com/user-attachments/assets/99cc878b-0630-49cc-8d88-6998bd5e05fe)



This PCB is fixed to the bottom side of the box by magnetic legs. It does perfectly fit and snaps on to the frame.
![IKEA-Hack - Wandleuchte OBEGRÄNSAD -02](https://github.com/user-attachments/assets/67e3881f-99cf-4a68-9966-ad558bcb073b)

The LDR (GL5528) is mounted to the case using a "5mm Clear LED Holder Clip for Bezel Mounting Panel Display".
![IKEA-Hack - Wandleuchte OBEGRÄNSAD -04](https://github.com/user-attachments/assets/a8861e9d-01bd-4b1c-b127-7b68af0e57a9)


> I've got mine just 2 weeks ago, but I've already fixed lots of bugs, cleaned up the code, as well as implemented new features and a number of new plugins. Will probably post it as an fork in a week or two, as the owner seemingly has lost interest in the project. It's more like a version `1.9`, so I don't expect it ever to be merged, given the state of existing PR's have died out.

I am eager to see your work and would appreciate posting this fork.

> I've also got it working on newer type of boards. ESP32-C3 for example wouldn't compile at all, but is now handling it all strong.

I used the WEMOS S3 mini ( https://www.wemos.cc/en/latest/s3/s3_mini.html ). This board (based ESP32-S3FH4R2) is pin compatible to the WEMOS D1 mini, so my PCB should run with D1 mini too (but did never test it).

edit: updated images of pcb

**Reply by JanPetterMG** [reply]
**The hole is actually 2,5 mm**, not 3 mm as first stated. *My original comment is updated/corrected too.*
I've finished a 2nd device (for a friend), went butter smooth this time. No destruction marks at all, it looks stock, except for some ~3 mm black screws with flat head, angled to dive in a bit, so that it becomes flat with the surface.

## Off topic: version 2.0 fork
I've mentioned posting a fork previously, I still plan to do, and my code has changed so much that it's definitively a version 2.0 if it ever gets merged.
Seems like the news has reached the current code owner, as quite a few patches arrived recently. I plan to backport them all into my own (planned) fork, before publishing it here on GitHub (and possibly opening an PR).

I actually had just the README section left on my todo-list, but now I feel I need a little bit more time, so that I can backport everything, and making some more key improvements with inspiration from suggestions from others.
When I post my fork, I plan to differentiate it from this original repository, it's not supposed to be an "alternative" path going forward, splitting the user base, NO !!! It will either be merged down the road, or be an superior and maintained continuation of the original repo.

### Highlights so far:
- 2x refresh rate - *animations is butter smooth now.*
- 4x gray scale resolution - *a lot of plugins has been updated to take advantage of this, looking so much better now.*
- 8 new plugins - *and I've gotten ideas for even more.*
- 2 new integrations - *new ways to control the device.*
- Fixed countless number of bugs - *including mayor improvements for a lot of the existing plugins.*
- Reduced number of lines - *the code base is smaller, but a lot more capable now.*
- Should be easier to develop new plugins - *due to less complexity.*
- Migrating any user-made custom plugins should be straight forward - *the changes are mostly under the hood, but some methods has changed a little bit, making it more user-friendly, flexible and robust.*
- Support for more devices - *no more compiling issues, take pretty much any board, and it will work out-of-the-box.*
- 4 MB flash memory - *still enough, and there's still some room for expansion.*
- Native OTA support - *no need for custom scripts.*
- Updated libraries - *up-to-date, using the latest version of everything.*

EDIT: Sorry for the delay, here it is: https://github.com/VIPnytt/Frekvens

**Reply by c64emulator** [reply]
any news so far?

**Reply by NemoN** [reply]
@JanPetterMG I am also very interested in your version...

**Comment by c64emulator** [comment]
> Removing the back was a bit tricky, so if I ever make another one, I think I'll cut a rectangular hole and then 3D print a cover for it. 

would recommend another topic for this thread. "hardware modding" would fit better.

**Comment by deepcoder** [comment]
While this might sound a bit violent and destructive, I found that using a hammer and a putty scraper or chisel with a fine edge removes each rivet with 2 or so quick and sharp blows. Hit the outside of the rivet toward the center of panel, not from the inside toward the edge. If you hit on the inside toward the edge you may break the cover around the rivet. While the rivets fly off around the lab this route, I found this much cleaner and better than the metal grindings created by drilling. Use a pair of pliers to remove any bends in the metal frame from this hammering.

I epoxied nylon flat head screws to the inside of 6  or so of the resulting holes at corners and hanging points, just long enough to allow placing a nut on the screw with the cover on. Then epoxied 4 rubber/silicon pads at the corner, each just taller than the protruding screws and nuts. This to protect the wall when hanging the unit.

Epoxying two of the same size nylon screws to the back (along the center line) of the ESP32 makes nice mating to the channel in the rail between each  of the four panels. It is just a friction push fit of the nylon screws into the channel. Mounting the ESP 32 on this rail near the connector soldering points and using a ribbon cable with plug connector (allowing the removal of the ESP 32) makes a clean mount.

**Reply by c64emulator** [reply]
pls provide some pictures - "One picture tells more than thousand words" ;)
Thanks!
